UOP1_Project/Assets/Scripts/Characters/EnemyColliderZoneController.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyColliderZoneController : MonoBehaviour
+{
+	[SerializeField] private VoidEventChannelSO _enterZone = default;
+
+	[SerializeField] private VoidEventChannelSO _exitZone = default;
+
+	private void OnTriggerEnter(Collider other)
+	{
+		if (other.GetComponent<Protagonist>() != null)
+		{
+			_enterZone.RaiseEvent();
+		}
+	}
+
+	private void OnTriggerExit(Collider other)
+	{
+		if (other.GetComponent<Protagonist>() != null)
+		{
+			_exitZone.RaiseEvent();
+		}
+	}
```

Comment:
This could be a very generic `ZoneTriggerController` if you detected the collision by layer:
``` csharp
[SerializeField] private LayerMask _layers = default;

private void OnTriggerEnter(Collider other)
{
	if (((1 << other.gameObject.layer) & _layers) != 0)
	{
		// stuff
	}
}
```